superset/connectors/sqla/models.py

```diff
@@ -780,8 +780,10 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
         # Database spec supports join-free timeslot grouping
         time_groupby_inline = db_engine_spec.time_groupby_inline
 
-        cols: Dict[str, Column] = {col.column_name: col for col in self.columns}
-        metrics_dict: Dict[str, SqlMetric] = {m.metric_name: m for m in self.metrics}
+        columns_by_name: Dict[str, TableColumn] = {
+            col.column_name: col for col in self.columns
+        }
+        metrics_by_name: Dict[str, SqlMetric] = {m.metric_name: m for m in self.metrics}
```

Comment:
Interesting that mypy didn't pick up on the incorrect typing of `cols`.

Reply:
Agreed.